images/sds-replicated-volume-controller/src/pkg/controller/linstor_resources_watcher.go

```diff
@@ -206,30 +232,113 @@ func ReconcileParams(
 	log.Info("[ReconcileParams] ends work")
 }
 
-func ReconcileTieBreaker(
+func ReconcilePVReplicas(
 	ctx context.Context,
 	log logger.Logger,
+	cl client.Client,
 	lc *lapi.Client,
 	rds map[string]lapi.ResourceDefinitionWithVolumeDefinition,
 	rgs map[string]lapi.ResourceGroup,
+	res map[string][]lapi.Resource,
+	pvs []core.PersistentVolume,
 ) {
-	log.Info("[ReconcileTieBreaker] starts work")
+	log.Info("[ReconcilePVReplicas] starts work")
 
-	allResources := make(map[string][]lapi.Resource, len(rds)*3)
-	for name := range rds {
-		res, err := lc.Resources.GetAll(ctx, name)
+	for i := range pvs {
+		pv := &pvs[i]
+		if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != PVCSIDriver {
+			continue
+		}
+
+		RGName := rds[pv.Name].ResourceGroupName
+		rg := rgs[RGName]
+		log.Debug(fmt.Sprintf("[ReconcilePVReplicas] PV: %s, RG: %s", pv.Name, rg.Name))
+
+		resources, ok := res[pv.Name]
+		if !ok {
+			continue
+		}
+
+		replicasErrLevel := checkPVMinReplicasCount(ctx, log, lc, rg, resources)
+
+		origLabelVal, exists := pv.Labels[pvNotEnoughReplicasLabel]
+		log.Debug(fmt.Sprintf("[ReconcilePVReplicas] Update label \"%s\", old: \"%s\", new: \"%s\"", pvNotEnoughReplicasLabel, origLabelVal, replicasErrLevel))
+
+		upd := false
+		if replicasErrLevel == "" && exists {
+			delete(pv.Labels, pvNotEnoughReplicasLabel)
+			upd = true
+		}
+		if replicasErrLevel != "" && replicasErrLevel != origLabelVal {
+			pv.Labels[pvNotEnoughReplicasLabel] = replicasErrLevel
+			upd = true
+		}
+
+		if upd {
+			err := UpdatePV(ctx, cl, pv)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("[ReconcilePVReplicas] unable to update the PV, name: %s", pv.Name))
+			}
+		}
+	}
+
+	log.Info("[ReconcilePVReplicas] ends work")
+}
+
+func checkPVMinReplicasCount(
+	ctx context.Context,
+	log logger.Logger,
+	lc *lapi.Client,
+	rg lapi.ResourceGroup,
+	resources []lapi.Resource,
+) string {
+	placeCount := int(rg.SelectFilter.PlaceCount)
+	upVols := 0
+
+	if placeCount <= 0 {
+		return ""
+	}
+
+	for _, r := range resources {
+		volList, err := lc.Resources.GetVolumes(ctx, r.Name, r.NodeName)
 		if err != nil {
-			log.Error(err, fmt.Sprintf("[ReconcileTieBreaker] unable to get Linstor Resources by the Resource Definition, name: %s", name))
+			log.Error(err, fmt.Sprintf("[checkPVMinReplicasCount] unable to get Linstor Resources Volumes, name: %s, node: %s", r.Name, r.NodeName))
 		}
 
-		allResources[name] = res
+		for _, v := range volList {
+			if v.State.DiskState == "UpToDate" {
+				upVols++
+			}
+		}
+	}
```

Comment:
Currently we're alerting about the number of `lapi.Resource` being too small. For me it would make more sense to alert about number of unique `NodeName` of `lapi.Resource` to be too small, i.e. when all resources are on the same node - it's bad.